### PR TITLE
crimson/os/seastore: consolidate seastore_t and seastore_cache logs

### PR DESCRIPTION
**Pull request by cyx1231st**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/seastore/cache.h [hunk 1]
```diff
@@ -253,11 +264,14 @@ class Cache {
     CachedExtentRef ret;
     LOG_PREFIX(Cache::get_extent_if_cached);
     auto result = t.get_extent(offset, &ret);
-    if (result != Transaction::get_extent_ret::ABSENT) {
-      // including get_extent_ret::RETIRED
-      SUBDEBUGT(seastore_cache,
-	"Found extent at offset {} on transaction: {}",
-	t, offset, *ret);
+    if (result == Transaction::get_extent_ret::RETIRED) {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cache.h:267]: should we return empty CachedExtentRef here? IIUC `TransactionManager::get_extent_if_live` and `SegmentCleaner::gc_reclaim_space()` should not rewrite retired extents.

File: src/crimson/os/seastore/cache.cc [hunk 2]
```diff
@@ -1289,21 +1340,27 @@ Cache::get_next_dirty_extents_ret Cache::get_next_dirty_extents(
...
+  } else {
+    DEBUGT("max_bytes={}B, seq={}, dirty_from={}",
+           t, max_bytes, seq, dirty.begin()->get_dirty_from());
+  }
   std::vector<CachedExtentRef> cand;
   size_t bytes_so_far = 0;
   for (auto i = dirty.begin();
        i != dirty.end() && bytes_so_far < max_bytes;
        ++i) {
     if (i->get_dirty_from() != journal_seq_t() && i->get_dirty_from() < seq) {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cache.cc:1355]: Seems to me dirty extents always need a valid dirty_from. If it is correct, should we instead assert that `i->get_dirty_from() != journal_seq_t()` is always true?

**Comment by cyx1231st** [comment]
changeset:
* add commit `crimson/os/seastore/journal: add logs to flush operation`
* refine commit titles

File: src/crimson/os/seastore/cache.cc [hunk 3]
```diff
@@ -613,7 +622,6 @@ void Cache::register_metrics()
 
 void Cache::add_extent(CachedExtentRef ref)
 {
-  LOG_PREFIX(Cache::add_extent);
```
Comment (athanatos) [inline src/crimson/os/seastore/cache.cc:616]: I'd sort of like to keep these, perhaps at TRACE?

File: src/crimson/os/seastore/cache.cc (hunk 2 above)
Comment (athanatos) [reply src/crimson/os/seastore/cache.cc:1355]: Yeah, I think that's right.

File: src/crimson/os/seastore/cache.cc (hunk 3 above)
Comment (cyx1231st) [reply src/crimson/os/seastore/cache.cc:616]: These are private methods, some are removed as they are instead logged by the caller and there are more information from the caller.

I'll revisit them, add comments if the removed log is coverred by caller, and keep the rest.

Comment (athanatos) [reply src/crimson/os/seastore/cache.cc:616]: Eh, if they're covered elsewhere that's fine.

**Review by athanatos — APPROVED** [review]

**Comment by cyx1231st** [comment]
changeset:
* https://github.com/ceph/ceph/pull/44649/commits/a3d1d22e01e4df85e2513f3193c05ae18cbf2deb:
  * reuse remove_extent in commit_retire_extent;
  * assert dirty_from is valid;
* https://github.com/ceph/ceph/pull/44649/commits/a07232e6818cbc693856f484f5391bd4068f3557:
  * keep trace log in add_extent() and remove_extent();

**Comment by cyx1231st** [comment]
jenkins retest this please

**Comment by cyx1231st** [comment]
rebased

**Pull request merged**